src/main/webapp/courseLoader.js

```diff
@@ -1,164 +1,205 @@
+    loadComments(dummyComments, /* isCourse */ false);
+
+    const tempHoursList = termDataObject.hoursList;
+    const hoursList = [
+      ["hours"],
+      /* dummyHourRating */ [3],
+      /* dummyHourRating */ [8],
+    ].concat(tempHoursList);
+
+    const hourData = new google.visualization.arrayToDataTable(hoursList);
+    const hourOptions = {
+      title: "Hours Spent per Week",
+      legend: { position: "none" },
+      vAxis: { title: "# Students" },
+      hAxis: { title: "Hours" },
+      histogram: {
+        hideBucketItems: true,
+      },
+    };
+    const hourChart = new google.visualization.Histogram(
+      document.getElementById("hours-chart")
+    );
+    hourChart.draw(hourData, hourOptions);
+
+    const tempDiffList = termDataObject.difficultyList;
+    const diffList = [
+      ["difficulty"],
+      /* dummyDifficultyRating */ [1],
+      /* dummyDifficultyRating */ [4],
+    ].concat(tempDiffList);
+    const diffData = new google.visualization.arrayToDataTable(diffList);
+    const diffOptions = {
+      title: "Difficulty of Class",
+      legend: { position: "none" },
+      vAxis: { title: "# Students" },
+      hAxis: { title: "Difficulty" },
+      histogram: {
+        hideBucketItems: true,
+      },
+    };
+    const diffChart = new google.visualization.Histogram(
+      document.getElementById("diff-chart")
+    );
+    diffChart.draw(diffData, diffOptions);
+
+    const tempTermPerceptionList = termDataObject.termPerceptionList;
+    const termPerceptionList = [
+      ["Term Perception"],
+      /* dummyPerceptionRating */ [11],
+      /* dummyPerceptionRating */ [5],
+    ].concat(tempTermPerceptionList);
+    const termPerceptionData = new google.visualization.arrayToDataTable(
+      termPerceptionList
+    );
+    const termPerceptionOptions = {
+      title: "Perception of Term Reviews",
+      legend: { position: "none" },
+      vAxis: { title: "Perception" },
+      hAxis: { title: "Comment Quantity" },
+      histogram: {
+        hideBucketItems: true,
+      },
+    };
+    const termPerceptionChart = new google.visualization.Histogram(
+      document.getElementById("term-chart")
+    );
+    termPerceptionChart.draw(termPerceptionData, termPerceptionOptions);
+
+    makeTermRatingChart();
+  }
+
+  async function makeTermRatingChart() {
+    const currentTerm = term.split(/(\s+)/).filter((e) => e.trim().length > 0);
+    let prevTerm1;
+    let prevTerm2;
+
+    if (currentTerm[0] == "Spring") {
+      prevTerm1 = `Fall ${currentTerm[1]}`;
+      prevTerm2 = `Spring ${String(parseInt(currentTerm[1] - 1))}`;
+    } else {
+      prevTerm1 = `Spring ${currentTerm[1]}`;
+      prevTerm2 = `Fall ${String(parseInt(currentTerm[1] - 1))}`;
+    }
+
+    const prevTermData1 = await getPreviousTermData(prevTerm1);
+    const prevTermData2 = await getPreviousTermData(prevTerm2);
+
+    const average = (list) =>
+      list.reduce((prev, curr) => prev + curr) / list.length;
+
```

Comment:
Add an initial value to this reduce statement (second arg with just 0 or w/e).
Otherwise it'll throw an error when you operate on an empty array